File: 5 занятие/4E.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long double ld;

ld v1, v2, a;

ld dist(ld x) {
    return sqrt((1.0 - a) * (1.0 - a) + x * x) / v1 +
           sqrt((1.0 - x) * (1.0 - x) + a * a) / v2;
}


int main() {
    cin >> v1 >> v2 >> a;

    ld curr = 0;
    ld step = 1.0;

    ld DX = 0.00000001;

    while (step > DX) {
        ld t1 = dist(curr + step);
        ld t2 = dist(curr + step + DX);
        if (t2 < t1) {
            curr += step; 
        } else {
            step /= 2;
        }
    }

    cout << curr;
}

File: 5 занятие/4E.py
v1, v2 = map(float, input().split())
a = float(input())
def dist(x):
    return ((1 - a) ** 2 + x**2) ** 0.5 / v1 + ((1 - x) ** 2 + a**2 )**0.5 / v2

curr = 0
step = 1
DX = 0.0000001

while step > DX:
    t1 = dist(curr + step)
    t2 = dist(curr + step + DX)
    if (t2 < t1):
        curr += step
    else:
        step /= 2
print(curr)
